core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -17,27 +17,62 @@
  * functions on $window to be mocked in unit tests.
  */
 
-oppia.factory('UrlService', ['$window', function($window) {
-  return {
-    getUrlParams: function() {
-      var params = {};
-      var parts = $window.location.href.replace(
-        /[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
-          params[key] = value;
+oppia.factory('UrlService', [
+  '$window', '$location',
+  function(
+      $window, $location) {
+    return {
+      // This function is for testing purposes (to mock $window.location)
+      getCurrentUrl: function() {
+        return $window.location;
+      },
+      getUrlParams: function() {
+        var params = {};
+        var parts = this.getCurrentUrl().href.replace(
+          /[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
+            params[key] = value;
+          }
+        );
+        return params;
+      },
+      isIframed: function() {
+        var pathname = this.getPathname();
+        var urlParts = pathname.split('/');
+        return urlParts[1] === 'embed';
+      },
+      getPathname: function() {
+        return this.getCurrentUrl().pathname;
+      },
+      getParentExplorationIds: function() {
+        if ($location.search().parent) {
+          if ($location.search().parent.constructor === Array) {
+            return $location.search().parent;
+          } else if ($location.search().parent) {
+          /* By default, single parent id gave a single string whereas, an array
+             with  a single element is what is required, hence the additional
+             condition */
+            return [$location.search().parent];
+          }
+        } else {
+          return null;
         }
-      );
-      return params;
-    },
-    isIframed: function() {
-      var pathname = this.getPathname();
-      var urlParts = pathname.split('/');
-      return urlParts[1] === 'embed';
-    },
-    getPathname: function() {
-      return $window.location.pathname;
-    },
-    getHash: function() {
-      return $window.location.hash;
-    }
-  };
-}]);
+      },
+      /* parameterList is an array of exploration ids from which 1 is popped out
+         and then, URL updated. */
+      updateParameterList: function(parameterList) {
+        var parameterString = '#?parent=';
+        for (var i = 0; i < parameterList.length - 1; i++) {
+          parameterString += parameterList[i] + '&';
+        }
+        return parameterString.slice(0, -1);
+      },
+      /* Use UrlService.pushParentIdToUrl(id); at the state which
+         redirects to push current exploration id to url stack. */
+      pushParentIdToUrl: function(explorationId) {
+        $location.search({parent: explorationId});
```

Comment:
Add line breaks after `({` and before `})`

Reply:
Done, removed this function